presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoPageSink.java

```diff
@@ -76,36 +75,36 @@
 {
     private final MongoSession mongoSession;
     private final ConnectorSession session;
-    private final SchemaTableName schemaTableName;
+    private final MongoTableHandle table;
     private final List<MongoColumnHandle> columns;
     private final String implicitPrefix;
 
     public MongoPageSink(
             MongoClientConfig config,
             MongoSession mongoSession,
             ConnectorSession session,
-            SchemaTableName schemaTableName,
+            MongoTableHandle table,
             List<MongoColumnHandle> columns)
     {
         this.mongoSession = mongoSession;
         this.session = session;
-        this.schemaTableName = schemaTableName;
+        this.table = table;
         this.columns = columns;
         this.implicitPrefix = requireNonNull(config.getImplicitRowFieldPrefix(), "config.getImplicitRowFieldPrefix() is null");
     }
 
     @Override
     public CompletableFuture<?> appendPage(Page page)
     {
-        MongoCollection<Document> collection = mongoSession.getCollection(schemaTableName);
+        MongoCollection<Document> collection = mongoSession.getCollection(table);
         List<Document> batch = new ArrayList<>(page.getPositionCount());
 
         for (int position = 0; position < page.getPositionCount(); position++) {
             Document doc = new Document();
 
             for (int channel = 0; channel < page.getChannelCount(); channel++) {
                 MongoColumnHandle column = columns.get(channel);
-                doc.append(column.getName(), getObjectValue(columns.get(channel).getType(), page.getBlock(channel), position));
+                doc.append(column.getName(), getObjectValue(column.getType(), page.getBlock(channel), position));
```

Comment:
I suppose that this change isn't mandatory to support case insensitive collection name. Please let me know if I am wrong.